README.md

```diff
@@ -611,3 +611,59 @@ status:
   binding:
     name: kafka-eftg9
 ```
+
+## Multi-Application Bindings
+
+The `ServiceBinding` resource is limited by the [Service Binding](#service-binding) spec to bind a single service by-name to a single application by-name. If the application's name is generated and not known in advance, the binding implicitly must be created after the application. While level-based reconciliation will converge on the desired state, the application will roll out initially without the bound service. Depending on the application this may cause a crash loop or a misconfiguration (e.g. using an in-memory database designed for development in a production environment). By switching the application reference from using a name to a label selector the service can be bound to the application resource as it is created.
+
+A `ServiceBinding` **MAY** define the application reference by-name or by-[label selector][ls]. A name and selector **MUST NOT** be defined in the same reference.
+
+As label selectors are inherently queries that return zero-to-many resources, implementations **SHOULD** handle each matching resource as if it was specified by name in a distinct `ServiceBinding` resource. Partial failures **SHOULD** be aggregated and reported on the binding status's `Ready` condition.
+
+[ls]: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  ...
+spec:
+  application:          # PodSpec-able resource ObjectReference-able
+    ...
+    name:               # string, optional, mutually exclusive with selector
+    selector:           # metav1.LabelSelector, optional, mutually exclusive with name
+  ...
+status:
+  ...
+```
+
+### Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name: online-banking-to-account-service
+spec:
+  name: account-service
+
+  application:
+    apiVersion: apps/v1
+    kind:       Deployment
+    selector:
```

Comment:
I like this sample - helps to visualize.    One thing in my mind is the conversation from #52 and whether we'll put all of the extensions in a single place or keep them in-line with the rest of the core spec.

Reply:
yea, I've been torn on that question. The `apiVersion` should be something custom until we make the decision to inline extensions into the standard resource structure.

Reply:
moved the example versions away from v1alpha1